### ISSUE TITLE
Number of samples and class memberships

### ISSUE THREAD

**Issue by vandres98** [helme/ecg_ptbxl_benchmarking#14]
Hi,
I am a little bit confused about the number of samples in the notebook **Finetuning-Example.ipynb.**
I see 21430 samples in total (train and validation together).
However, the paper [physionet](https://physionet.org/content/ptb-xl/1.0.1/) argues that there are 21837 records and thats also what I see in the data/ptbxl/records100 folder.
Why are there 407 records missing? 

Other question: How can I interpret the label sets y_train and y_val?
Which representation (10000,01000,00100,00010,00001) correspond to which class (normal, MI, STTC, CD, HYP)?
I cannot map them according to the numbers of samples in the classes because they don't match.

Can you help me and clear that up? Thank you very much!

**Comment by helme** [comment]
Hi @vandres98 ,
thanks for your interest in our work. 

1. there are only 21430 samples in case of "superdiagnostic", since there are 407 samples having no diagnostic statement at all. In `utils.select_data` we filter those samples to ensure at least one diagnostic label. I hope this answers your first question.
![label_counts](https://user-images.githubusercontent.com/2674825/130439008-beee78b8-acb4-40f8-905f-57332f93b160.png)

2. for corresponding class labels you can use the fourth returning argument from `utils.select_data` which is an instance of  [MultiLabelBinarizer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html) having an attribute `classes_` containing the ordering of classes as list. This instance is also stored in `output/mlb.pkl` as a pickle-file.

I hope this answers your questions ;)

Best
helme

**Comment by vandres98** [comment]
HI helme,

thank you for the answer!

1: Thank you that makes sense! I am confused about your class count though. 
The ptb-xl paper states the following memberships:
![Bildschirmfoto von 2021-08-23 16-40-57](https://user-images.githubusercontent.com/66438051/130467055-af0ba696-da42-4061-a393-897b9d3d85a7.png)
Because it's multilabel, the sum of the amounts is of course greater than the sample-size.

2. I got the y data with the binarized labels already from your Finetuning-Example.ipynb. However, I have troubles assigning them to the (medical) classes like in the chapter above. Could you provide the connections please? That would be a huge help!

Thank you and best regards
Viktoria

**Comment by helme** [comment]
Hi @vandres98 ,

1. This was not about class counts, but rather "number of labels per sample"-count, i.e. `superdiagnostic_len` codes the number of labels associated to each sample (multi label). So most samples (16272) have one label, 4079 samples have two diagnostic labels etc.
2. as already explained, there is a instance of MultiLabelBinarizer returned in `utils.select_data` which has an attribute `classes_`. Alternatively you can load the pickle file (stored in `/output/mlb.pkl`). E.g.:
![screenshot](https://user-images.githubusercontent.com/2674825/130766068-755b2279-556b-42ca-b488-a75612bbd53e.png)

so` ['CD', 'HYP', 'MI', 'NORM', 'STTC']` is the ordering in this case, e.g. `[0,0,0,1,0]` is sample associated with diagnostic class `'NORM'`. 

I hope this answers your questions.

Best,
@helme
